src/session_impl.cpp

```diff
@@ -5548,6 +5554,28 @@ namespace aux {
 			, boost::ref(m_alerts), _1, cb), salt);
 	}
 
+	void session_impl::dht_get_peers(sha1_hash const& info_hash)
+	{
+		if (!m_dht) return;
+		m_dht->get_peers(info_hash, boost::bind(&on_dht_get_peers, boost::ref(m_alerts), info_hash, _1));
+	}
+
+	void session_impl::dht_announce(sha1_hash const& info_hash, int port, int flags)
```

Comment:
can you describe again what the use case is for pretending to be on a torrent (that isn't loaded in libtorrent)?